src/ui/components/DevToolsProcessingScreen.tsx

```diff
@@ -45,5 +81,43 @@ export function DevToolsProcessingScreen() {
     message = `Processing... (${Math.round(processingProgress)}%)`;
   }
 
-  return <LoadingScreen message={message} secondaryMessage={secondaryMessage} />;
+  const onChange = async (event: ChangeEvent<HTMLInputElement>) => {
+    if (event.target.checked) {
+      const granted = await requestNotificationPermission();
+      if (!granted) {
+        event.target.checked = false;
+        event.target.blur();
+      }
+    }
+  };
+
+  return (
+    <LoadingScreen
+      message={
+        <>
+          <div className={styles.Message}>{message}</div>
+          <div className={styles.SecondaryMessage}>{secondaryMessage}</div>
+          {notificationSupported && (
+            <label
+              className={styles.NotifyMessage}
+              data-disabled={notificationPermission === "denied" || undefined}
+              title={
+                notificationPermission === "denied"
+                  ? "Notifications have been disabled in browser settings"
+                  : undefined
+              }
+            >
+              <input
+                disabled={notificationPermission === "denied"}
+                className={styles.Checkbox}
+                onChange={onChange}
+                type="checkbox"
+              />{" "}
+              Notify me when this tab loads
```

Comment:
I wonder if people wouldn't expect to get a **push** notification here but we don't support this. Maybe we should somehow warn them that they can't close this tab to get those?